cellpack/autopack/Environment.py

```diff
@@ -477,6 +298,82 @@ def setSeed(self, seedNum):
         self.seed_set = True
         self.seed_used = SEED
 
+    def _prep_ingredient_info(self, composition_info, ingredient_name=None):
+        objects_dict = self.recipe_data["objects"]
+        object_key = composition_info["object"]
+        base_object = objects_dict[object_key]
+        ingredient_info = deep_merge(copy.deepcopy(base_object), composition_info)
+        ingredient_info["name"] = (
+            ingredient_name if ingredient_name is not None else object_key
+        )
+        return ingredient_info
+
+    def _step_down(self, compartment_key):
+        composition_dict = self.recipe_data["composition"]
+        compartment = self.create_compartment(compartment_key)
+        compartment_info = composition_dict[compartment_key]
+        for region_name, obj_keys in compartment_info.get(
+            "regions", {}
+        ).items():  # check if entry in compositions has regions
+            recipe = Recipe(name=f"{compartment_key}_{region_name}")
+            for key_or_dict in obj_keys:
+                if not isinstance(key_or_dict, dict):
+                    key = key_or_dict
+                    if key in self.compartment_keys:
+                        # deal with compartment
+                        return self._step_down(key)
```

Comment:
I think we can just call `_step_down` without returning, to allow for the rest of the internal compartments to process (may need to verify with debugger)